packages/brokeneck-fastify/lib/plugins/auth/auth0/provider.js

```diff
@@ -37,10 +37,21 @@ function Auth0Provider(options, logger) {
 
       return user
     },
-    async listGroups() {
-      const groups = await auth0.getRoles()
+    async listGroups({ pageNumber, pageSize }) {
+      const page = pageNumber ? Number(pageNumber) - 1 : 0
 
-      logger.debug({ groups })
+      const data = await auth0.getRoles({
+        page: page,
+        per_page: pageSize,
+        include_totals: true
+      })
+
+      const groups = {
+        data: data.roles,
+        nextPage: data.roles.length === data.limit ? (page + 2).toString() : ''
+      }
```

Comment:
some of this pagination logic is exactly the same for users and will be the same if we paginate elsewhere too, so I would consider extracting it or creating some helper functions